crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -843,6 +857,56 @@ impl AuthorityPerEpochStore {
         self.finish_consensus_transaction_process_with_batch(write_batch, key, consensus_index)
     }
 
+    /// Called when a ceritified checkpoint is inserted into the checkpoint store.
+    /// This function records the validators' participation in the certified checkpoint by
+    /// incrementing the counters stored in the epoch tables. Tallying score metrics are
+    /// also updated at this point to send the most up-to-date scores.
+    pub fn record_certified_checkpoint_signatures(
+        &self,
+        certified_checkpoint: &CertifiedCheckpointSummary,
+    ) -> Result<(), TypedStoreError> {
+        // TODO: what to do when the committee is not valid and an error is returned?
+        if let Ok(signed_authorities) = certified_checkpoint
+            .signatory_authorities(&self.committee)
+            .collect::<Result<Vec<&AuthorityName>, _>>()
+        {
+            debug!(
+                "Recording certified checkpoint signatures from {:?}",
+                signed_authorities
+            );
+
+            let old_values = self
+                .tables
+                .num_certified_checkpoint_signatures
+                .multi_get(signed_authorities.clone())?;
+
+            let new_values = signed_authorities
+                .into_iter()
+                .zip(old_values.into_iter().map(|v| v.unwrap_or(0) + 1)); // increment counter
+
+            // Send tallying rule score through prometheus.
+            new_values.clone().for_each(|(authority_name, score)| {
```

Comment:
The clone here will essentially duplicate `signed_authorities` vector and repeat `zip` logic from above twice.
Probably more reasonable to just convert `new_values` into vector before this line, and avoid cloning it.